src/templates/InstantSearch.js/src/app.js.hbs

```diff
@@ -34,24 +31,28 @@ search.addWidget(
   {{/each}}
 </article>
 `,
-    },
-    {{/if}}
-  })
-);
+  },
+  {{/if}}
+});
 
 {{#each attributesForFaceting}}
-search.addWidget(
-  instantsearch.widgets.refinementList({
-    container: '#{{this}}-list',
-    attribute: '{{this}}',
-  })
-);
+const {{this}}RefinementList = instantsearch.widgets.refinementList({
+  container: '#{{this}}-list',
+  attribute: '{{this}}',
+});
 
 {{/each}}
-search.addWidget(
-  instantsearch.widgets.pagination({
-    container: '#pagination',
-  })
-);
+const pagination = instantsearch.widgets.pagination({
+  container: '#pagination',
+});
+
+search.addWidgets([
+  searchBox,
+  hits,
```

Comment:
Do we really have to store them? We can easily create anonymous widget.

Reply:
Sounds good – going to be easier than getting Handlebars to work properly in this environment.